### DISCUSSION TITLE
realtime subscription events for available state of records

### DISCUSSION THREAD

**Discussion by m-2k** [pocketbase/pocketbase#4592] [Q&A] [answered]
how can a client-side know that a `record` is no longer available for reading due to API rules (`@request.auth.id ?= members.id`)

current client-side code:
```js
return await pocketbase.collection('my_table').subscribe('*', ({ action, record }) => {
  switch (action) {
  ...
  }
```

**Comment by ganigeorgiev** [comment, marked as answer]
Your issue seems similar to https://github.com/pocketbase/pocketbase/issues/2522.

It is a common request but we currently don't emit events when a record become unavailable/inaccessible mostly due to eventual performance concerns (_I'll experiment with it during the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355)_).

**Reply by m-2k** [reply]
>I'll experiment with it during the refactoring

yay, cool
